### PR TITLE
feat: apply coupon to circle subscription

### PR DESCRIPTION
**Pull request by Carolusian**

### CODE REVIEW

**Review by devformatters — APPROVED** [review]
Let's try it. 🖇

**Review by guoliu — APPROVED** [review]

File: src/connectors/paymentService.ts [hunk 1]
```diff
@@ -695,11 +696,21 @@ export class PaymentService extends BaseService {
...
     const { userId, priceId, providerCustomerId, providerPriceId } = data
 
+    const ivt = await this.findPendingInvitation({ userId, priceId })
     // Create from Stripe
-    const stripeSubscription = await this.stripe.createSubscription({
-      customer: providerCustomerId,
-      price: providerPriceId,
-    })
+    let stripeSubscription
+    if (isUndefined(ivt)) {
```
Comment (robertu7) [inline src/connectors/paymentService.ts:702]: it's ok to just `ivt` instead of `isUndefined(ivt)`

**Review by robertu7 — APPROVED** [review]

File: src/connectors/paymentService.ts [hunk 2]
```diff
@@ -760,5 +786,50 @@ export class PaymentService extends BaseService {
...
+  findPendingInvitation = async (params: {
+    userId: string
+    priceId: string
+  }) => {
+    const records = await this.knex
+      .select('ci.id', 'cc.provider_coupon_id')
+      .from('circle_invitation as ci')
+      .join('circle_price as cp', 'cp.circle_id', 'ci.circle_id')
+      .join('circle_coupon as cc', 'cc.id', 'ci.coupon_id')
+      .where({
```
Comment (robertu7) [inline src/connectors/paymentService.ts:813]: Since we allow circle owners to invite via email, we should find invitation via `user.email` also?

**Review by robertu7 — CHANGES_REQUESTED** [review]

File: src/connectors/paymentService.ts (hunk 2 above)
Comment (Carolusian) [reply src/connectors/paymentService.ts:813]: merged for the time being, I will have another PR for this.

Comment (Carolusian) [reply src/connectors/paymentService.ts:817]: @robertu7 please see the updated where clause for email checking

**Review by Carolusian — COMMENTED** [review]
@robertu7 I removed `isUndefined` from paymentService. But opted to keep it in stripe/index.ts as it seems more explicit in optional parameter checking.

**Review by robertu7 — APPROVED** [review]

File: src/connectors/paymentService.ts (hunk 2 above)
Comment (devformatters) [reply src/connectors/paymentService.ts:817]: Nice point. 💪🏻

**Pull request merged**